src/backend/distributed/executor/multi_executor.c

```diff
@@ -66,7 +66,8 @@ static CustomExecMethods RouterSingleModifyCustomExecMethods = {
 	.ExplainCustomScan = CitusExplainScan
 };
 
-static CustomExecMethods RouterMultiModifyCustomExecMethods = {
+/* not static to enable swapping in multi-modify logic during router execution */
+CustomExecMethods RouterMultiModifyCustomExecMethods = {
```

Comment:
Hm. I'm not sure that's correct, at least considering deadlock detection-adjacent features. I was led to believe that those changes were going to permit arbitrary multi-shard/multi-statement transactions. In that case, we want to (a) use the multi-task executor (and its transaction machinery) _even_ in cases where the modification hits only a single shard, or (b) use the single-shard path for single-shard modifications outside of a transaction block.

Does that make sense? I'm hesitant to make the suggested change until the other case is cleared up.